libs/velo-external-db-core/src/service/schema_aware_data.ts

```diff
@@ -112,4 +112,10 @@ export default class SchemaAwareDataService {
         const fields = _fields ?? await this.schemaInformation.schemaFieldsFor(collectionName)
         return fields.map((f: { field: any }) => f.field)
     }
+
+    private async projectionFor(collectionName: string, _projection?: string[]) {
+        const schemaContainsId = (await this.schemaInformation.schemaFieldsFor(collectionName)).some(f => f.field === '_id')
+        const projection = _projection ?? (await this.schemaInformation.schemaFieldsFor(collectionName)).map(f => f.field) 
```

Comment:
I would create a constant to the `schemaFieldsFor` result, and then I would use it where needed insted of calling the same function twice (I know that it uses cache and not really sending queries to the DB )